include/s3select_oper.h

```diff
@@ -1268,28 +1403,50 @@ class logical_operand : public base_statement
     return std::string("#");//TBD
   }
   virtual value& eval()
-  {bool res;
+  {
+    bool res;
+    value a = l->eval();
+    value b = r->eval();
```

Comment:
it's not optimized.
before AND/OR actual calculation, both side are evaluated. 
only the left side should be evaluated.